task3.py

```diff
@@ -0,0 +1,322 @@
+    """
+    Stores data into CSV files within corresponding folders
+
+    Parameters:
+        destination_folder (str): The path to the destination folder where the CSV files will be stored
+        folder_structure (dict): A nested dictionary representing the folder structure
+    """
+    for decade, countries in folder_structure.items():
+        for country, users in countries.items():
+            country_folder = os.path.join(destination_folder, decade, country)
+
+            max_age = calculate_max_age(users)
+            avg_registered_years = calculate_avg_registered_years(users)
+            popular_id = calculate_popular_id(users)
+
+            filename = f'max_age_{max_age}_avg_registered_{avg_registered_years:.2f}_popular_id_{popular_id}.csv'
+            file_path = os.path.join(country_folder, filename)
+
+            with open(file_path, 'w', newline='', encoding='utf-8') as csv_file:
+                fieldnames = users[0].keys()
+                writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+                writer.writeheader()
+                writer.writerows(users)
+
+            logger.info(f'Created file: {file_path}')
+
+
+def remove_decades_before_1960(folder_structure):
+    """
+    Removes decades before the 1960s from the folder structure
+
+    Parameters:
+        folder_structure (dict): A nested dictionary representing the folder structure
+    """
+    decades_to_remove = [decade for decade in folder_structure.keys() if int(decade[:2]) < 60]
+
+    for decade in decades_to_remove:
+        del folder_structure[decade]
+
+
+def log_structure_folder(destination_folder):
+    """
+    Logs the folder structure
+
+    Parameters:
+        destination_folder (str): The path to the destination folder whose structure will be logged
+    """
+    for foldername, subfolders, filenames in os.walk(destination_folder):
+        logger.info(f'Folder: {os.path.relpath(foldername, destination_folder)}')
+        for filename in filenames:
+            logger.info(f'\t- File: {filename}')
+
+
+def archive_destination_folder(destination_folder):
+    """
+    Archives the destination folder into a zip file
+
+    Parameters:
+        destination_folder (str): The path to the destination folder to be archived
+
+    Returns:
+        str: The path to the created zip file
+    """
+    zip_filename = f'{destination_folder}.zip'
+    with zipfile.ZipFile(zip_filename, 'w', zipfile.ZIP_DEFLATED) as zipf:
+        for foldername, subfolders, filenames in os.walk(destination_folder):
+            for filename in filenames:
+                filepath = os.path.join(foldername, filename)
+                zipf.write(filepath, os.path.relpath(filepath, destination_folder))
+
+    return zip_filename
+
+
+def main():
+    """
+    Main function that orchestrates the data preparation process
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('destination_folder', help='Path to the output folder')
```

Comment:
read the task carefuly:
•	Destination folder -- path to a folder where output file is going to be placed (required)

Reply:
make the argument **required**